iroh-api/src/p2p.rs

```diff
@@ -56,9 +85,69 @@ impl P2p {
             .await
             .map_err(|e| map_service_error("p2p", e))
     }
+
+    /// Get a stream of [`NetworkEvent`].
+    pub async fn network_events(&self) -> Result<BoxStream<'static, Result<NetworkEvent>>> {
+        let stream = self
+            .client
+            .network_events()
+            .await
+            .map_err(|e| map_service_error("p2p", e))?;
+        Ok(stream.boxed())
+    }
+
+    /// Subscribe to a Gossipsub Topic
+    ///
+    /// Gossipsub is a pub/sub protocol. This will subscribe you to a Gossipsub
+    /// topic. All Gossipsub messages for topics that you are subscribed to can
+    /// be read off the the `NetworkEvent` stream, as
+    /// `NetworkEvent::Gossipsub(GossipsubEvent::Message)`. You can access this
+    /// stream using `P2p::network_events`.
+    ///
+    /// There is currently no `gossipsub_unsubscribe` exposed to `iroh-api` crate
+    /// yet.
+    ///
+    /// Learn more about the Gossipsub protocol in the `libp2p-gossipsub`
+    /// [documentation](https://docs.rs/libp2p-gossipsub/latest/libp2p_gossipsub/).
+    //
+    // TODO(ramfox): write `gossipsub_messages(topic: String)` method that returns
+    // a stream of only the gossipsub messages on that topic
+    pub async fn gossipsub_subscribe(&self, topic: String) -> Result<bool> {
+        let topic = TopicHash::from_raw(topic);
+        self.client
+            .gossipsub_subscribe(topic)
+            .await
+            .map_err(|e| map_service_error("p2p", e))
+    }
+
+    /// Publish a message on a Gossipsub Topic.
+    ///
+    /// This allows you to publish a message on a given topic to anyone in your network that is
+    /// subscribed to that topic.
+    ///
+    /// Read the [`gossipsub_subscribe`][`crate::p2p::P2p::gossipsub_subscribe`] documentation for how to subscribe and receive
```

Comment:
this is failing:
```
warning: public documentation for `gossipsub_publish` links to private item `crate::p2p::P2p::gossipsub_subscribe`
   --> iroh-api/src/p2p.rs:123:5
    |
123 | /     /// Publish a message on a Gossipsub Topic.
124 | |     ///
125 | |     /// This allows you to publish a message on a given topic to anyone in your network that is
126 | |     /// subscribed to that topic.
127 | |     ///
128 | |     /// Read the [`gossipsub_subscribe`][`crate::p2p::P2p::gossipsub_subscribe`] documentation for how to subscribe and receive
129 | |     /// Gossipsub messages.
    | |___________________________^
    |
    = note: `#[warn(rustdoc::private_intra_doc_links)]` on by default
    = note: the link appears in this line:

            Read the [`gossipsub_subscribe`][`crate::p2p::P2p::gossipsub_subscribe`] documentation for how to subscribe and receive
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this link resolves only because you passed `--document-private-items`, but will break without
```
I'm guessing it's because the `p2p` module is technically private, even though the `P2p` struct itself is public?

Any thoughts on how to fix? 

Make the `p2p` mod public? Just silence the warning? Or is there a different way to reference this that makes it clear to rustdoc that it's fine? Using [`P2p::gossipsub_subscribe`] also triggered this warning.

Reply:
Oh no!  The mocks strike again!  The CI uses `--all-features` for this... which then breaks due to the mocks which seem to have different visibility then the real things.  Invoking `cargo doc` without `--all-features` makes it work as expected (and you can use the simpler ``[`P2p::gossipsub_subscribe`]`` form).

Unless I manage to get rid of the mocks before you're awake I suggest to just write it without linking for now to get around this: `` `P2p::gossipsub_subscribe` ``

Reply:
Try merging master and see if this is fixed, I hope it is!